### DISCUSSION TITLE
What engine should i use instead of pynio engine in xarray to read .hdf file

### DISCUSSION THREAD

**Discussion by Jayayadav33** [pydata/xarray#7980] [General] [answered]
I am trying to read a .hdf file. But there is an error for the line:
```python
data = xr.open_dataset(filename, engine='pynio')
```

```python-traceback

ValueError: unrecognized engine pynio must be one of: ['netcdf4', 'scipy', 'rasterio', 'store']

```

I am using windows 11 and unable to install pynio in anaconda. What engine should i use instead of pynio in the above command??

**Comment by andersy005** [comment, marked as answer]
@Jayayadav33, have you tried 

```python
data = xr.open_dataset(filename, engine='netcdf4')
```

or install `h5netcdf`  via `conda install -c conda-forge h5netcdf`


and use 

```python
data = xr.open_dataset(filename, engine='h5netcdf')
```

**Reply by kmuehlbauer** [reply]
Another try, depending on the source data (eg. HDF4) is `engine=rasterio`. You would need to install `rasterio`/`rioxarray`.

**Reply by Jayayadav33** [reply]
The hdf file is obtained from AERIS/ICARE and the filename is _DARDAR-CLOUD_v2.1.1_2017002114852_56830.hdf_
**data = xr.open_dataset(filename, engine='h5netcdf')**
This did not work.

**data = xr.open_dataset(filename, engine='netcdf4')**
This worked.

With engine = rasterio. Got the following error:
**RioXarrayError: Multiple resolution sets found. Use 'variable' or 'group' to filter.**
